### ISSUE TITLE
`lintr::lint(text=)` no longer reads personal settings

### ISSUE THREAD

**Issue by bastistician** [r-lib/lintr#2847]
(This doesn't affect a release yet, but is an issue of the recently merged #2805, raised here per request of @MichaelChirico.) 

The recent change from #2805 means that `~/.lintr` settings are no longer read when linting text input (`inline_data`), e.g., `lintr::lint(<R-code-as-string>)`, which is what Emacs-ESS uses to lint code in R buffers. So with the current development version of **lintr** I'd only see notes from the `default_linters`.

Here is an example where the non-default `duplicate_argument_linter()` is added in the settings:

```R
library("lintr")
cat("linters: linters_with_defaults(duplicate_argument_linter())\n", file = tf <- tempfile())
options("lintr.linter_file" = tf)
lint(text = "list(x = 1, x = 2)")  # parse_settings is TRUE by default
## i No lints found.
```

In **lintr** 3.2.0, this did produce a "lint".

**Comment by MichaelChirico** [comment]
Thanks for the report. I think in my head I did this intentionally with the thought "of course you shouldn't expect settings to be picked up if you're using `text=`", so of course this goes and proves me wrong :)

I'm not very familiar with Emacs/ESS backend, and a quick Google was no help -- why isn't Emacs just doing `lint(<file>)` instead of `lint(text=)`?

**Comment by MichaelChirico** [comment]
I see here it's Flymake doing the linting:

https://github.com/emacs-ess/ESS/blob/0eb240bcb6d0e933615f6cfaa9761b629ddbabdd/lisp/ess-r-flymake.el

But I can't find Flymake's own code.

**Comment by bastistician** [comment]
> why isn't Emacs just doing `lint(<file>)` instead of `lint(text=)`?

Buffers aren't necessarily visiting files. For those that do, linting should use the (edited) buffer content not what has been written (back) to the file. I'm no Emacs expert either but I guess that must be the reason.

**Comment by MichaelChirico** [comment]
cc (long-ago) original authors of the lintr-emacs integration: @vspinu and @jabranham

**Comment by MichaelChirico** [comment]
Here's the actual invocation of `esslint()`:

https://github.com/emacs-ess/ESS/blob/0eb240bcb6d0e933615f6cfaa9761b629ddbabdd/lisp/ess-r-flymake.el#L201-L216

And its definition:

https://github.com/emacs-ess/ESS/blob/0eb240bcb6d0e933615f6cfaa9761b629ddbabdd/lisp/ess-r-flymake.el#L84-L96

So that effectively Emacs is calling `lint()` like `lint(readChar(file), <buffer-point-min>, <buffer-point-max>)`.

I requested a patch to ESS to use `text=` instead which is definitely more appropriate for this use case, as well as explicitly supplying `parse_settings=TRUE`:

https://github.com/emacs-ess/ESS/pull/1325
